release/nightly_tests/dataset/read_benchmark.py

```diff
@@ -0,0 +1,115 @@
+import os
+import random
+import shutil
+import tempfile
+from typing import List, Optional, Tuple
+
+from PIL import Image
+
+import ray
+from ray.data.dataset import Dataset
+from ray.data.datasource import ImageFolderDatasource
+
+from benchmark import Benchmark
+
+
+def read_images(
+    root: str, size: Optional[Tuple[int, int]] = None, mode: Optional[str] = None
+) -> Dataset:
+
+    return ray.data.read_datasource(
+        ImageFolderDatasource(), root=root, size=size, mode=mode
+    )
+
+
+def generate_images(
+    num_images: int, sizes: List[Tuple[int, int]], modes: List[str], formats: List[str]
+) -> str:
+
+    dimensions = []
+    for mode in modes:
+        if mode in ["1", "L", "P"]:
+            dimension = 1
+        elif mode in ["RGB", "YCbCr", "LAB", "HSV"]:
+            dimension = 3
+        elif mode in ["RGBA", "CMYK", "I", "F"]:
+            dimension = 4
+        else:
+            raise ValueError(f"Found unknown image mode: {mode}.")
+        dimensions.append(dimension)
+
+    images_dir = tempfile.mkdtemp()
+
+    for image_idx in range(num_images):
+        size = random.choice(sizes)
+        file_format = random.choice(formats)
+        mode_idx = random.randrange(len(modes))
+        mode = modes[mode_idx]
+        dimension = dimensions[mode_idx]
+
+        width, height = size
+        file_name = f"{images_dir}/{image_idx}.{file_format}"
+        pixels_per_dimension = []
+        for _ in range(dimension):
+            pixels = os.urandom(width * height)
+            pixels_per_dimension.append(pixels)
+
+        image = Image.new(mode, size)
+        if len(pixels_per_dimension) == 1:
+            image.putdata(pixels_per_dimension[0])
+        else:
+            image.putdata(list(zip(*pixels_per_dimension)))
+        image.save(file_name)
+
+    return images_dir
+
+
+def run_images_benchmark(benchmark: Benchmark):
+    test_input = [
+        generate_images(100, [(256, 256)], ["RGB"], ["jpg"]),
```

Comment:
Those images are written to local fs, would it have any issue to run in a cluster setup?

Reply:
@jianoaix - yeah, I am changing the benchmark to be single node for now.

Reply:
We need to support distributed read though. We probably need to write them out to S3 in followup.

Reply:
Yes, plan to do it later as followup.